iceoryx_posh/test/moduletests/test_version_info.cpp

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/version/version_info.hpp"
+#include "iceoryx_versions.hpp"
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::version;
+using namespace iox::cxx;
+
+namespace
+{
+class VersionInfoSut : public VersionInfo
```

Comment:
This is a hidden white box test which we should avoid. I suggest an approach where we do not inherit from `VersionInfo` and make all the private members public via getter methods. The user cannot access them! Why should then the test access them?

We should only test the things which are available for the user. These are the comparision operators, serialization conversion and checkCompatibility and with that we have test cases like:

1. Two VersionInfos created with the same parameters should be equal
2. VersionInfos where only versionMajor is different should not be equal
2.1 also separate tests for versionMinor, versionPatch, versionTweak, buildDateString, commitIdString
3. We create one VersionInfo by serialization and one with parameters.
3.1 In separate tests, both VersionInfos have the same parameters and should be equal
3.2 the VersionInfos have different parameters and should not be equal
4. Create VersionInfo A with a random set of parameters, serialize it and create another VersionInfo B with the serialization. Create a third VersionInfo C with the same random set of parameters as A. Now create a serialization of B and C. The serialization should be equal.
4.1 The same but with a different random set of parameters.

I think those test cases should cover all the use cases we have. And for this we do not need to access the private members of this class.